### PR TITLE
Added petals models

### PR DESCRIPTION
**Pull request by filopedraz**
### Description

Added Petals services

- [x] Stable Beluga 2
- [x] Code Llama 34B

Missing

- [ ] Falcon 180B

### CODE REVIEW

**Comment by biswaroop1547** [comment]
LGTM, do we plan to add something on the daemon side to choose swarm or run it in petals swarm by default?

**Review by biswaroop1547 — APPROVED** [review]

File: cht-petals/build.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+set -e
+export VERSION=1.0.4
+source "$(dirname "${BASH_SOURCE[0]}")/../utils.sh"
+
+build_cpu ghcr.io/premai-io/chat-codellama-34b-cpu   codellama-34b   ${@:1}
+build_cpu ghcr.io/premai-io/chat-stable-beluga-2-cpu stable-beluga-2 ${@:1}
```
Comment (casperdcl) [inline cht-petals/build.sh:7]: meta-question:

is "the Petals way to use your local GPU" to separately connect it to Petals swarm?
i.e. there are no advantages to `build_gpu`? there's no way to force using both the local GPU and the swarm (without connecting local GPU to the swarm)?

Comment (casperdcl) [inline cht-petals/build.sh:7]: both `codellama-34b` and `stable-beluga-2` result in this error:

```python
 => ERROR [linux/amd64 8/9] RUN python3 download.py --model codellama-34b                                             13.4s
------
 > [linux/amd64 8/9] RUN python3 download.py --model codellama-34b:
12.76 Downloading model codellama-34b
12.76 Traceback (most recent call last):
12.76   File "/usr/local/lib/python3.10/site-packages/huggingface_hub/utils/_errors.py", line 261, in hf_raise_for_status
12.76     response.raise_for_status()
12.76   File "/usr/local/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
12.76     raise HTTPError(http_error_msg, response=self)
12.76 requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://huggingface.co/codellama-34b/resolve/main/tokenizer.model
12.76
12.76 The above exception was the direct cause of the following exception:
12.76
12.76 Traceback (most recent call last):
12.76   File "/usr/local/lib/python3.10/site-packages/transformers/utils/hub.py", line 429, in cached_file
12.76     resolved_file = hf_hub_download(
12.76   File "/usr/local/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py", line 118, in _inner_fn
12.76     return fn(*args, **kwargs)
12.76   File "/usr/local/lib/python3.10/site-packages/huggingface_hub/file_download.py", line 1195, in hf_hub_download
12.76     metadata = get_hf_file_metadata(
12.76   File "/usr/local/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py", line 118, in _inner_fn
12.76     return fn(*args, **kwargs)
12.76   File "/usr/local/lib/python3.10/site-packages/huggingface_hub/file_download.py", line 1541, in get_hf_file_metadata
12.76     hf_raise_for_status(r)
12.76   File "/usr/local/lib/python3.10/site-packages/huggingface_hub/utils/_errors.py", line 293, in hf_raise_for_status
12.76     raise RepositoryNotFoundError(message, response) from e
12.76 huggingface_hub.utils._errors.RepositoryNotFoundError: 401 Client Error. (Request ID: Root=1-6509fa14-4297b14d7ed8ad483f143d72;b922019e-fa84-4a98-b504-cdd3a3293c43)
12.76
12.76 Repository Not Found for url: https://huggingface.co/codellama-34b/resolve/main/tokenizer.model.
12.76 Please make sure you specified the correct `repo_id` and `repo_type`.
12.76 If you are trying to access a private or gated repo, make sure you are authenticated.
12.76 Invalid username or password.
12.76
12.76 The above exception was the direct cause of the following exception:
12.76
12.76 Traceback (most recent call last):
12.76   File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 382, in __call__
12.76     result = fn(*args, **kwargs)
12.76   File "/usr/src/app/download.py", line 17, in download_model
12.76     _ = LlamaTokenizer.from_pretrained(args.model)
12.76   File "/usr/local/lib/python3.10/site-packages/transformers/tokenization_utils_base.py", line 1813, in from_pretrained
12.76     resolved_vocab_files[file_id] = cached_file(
12.76   File "/usr/local/lib/python3.10/site-packages/transformers/utils/hub.py", line 450, in cached_file
12.76     raise EnvironmentError(
12.76 OSError: codellama-34b is not a local folder and is not a valid model identifier listed on 'https://huggingface.co/models'
12.76 If this is a private repository, make sure to pass a token having permission to this repo either by logging in with `huggingface-cli login` or by passing `token=<your_token>`
```

Comment (casperdcl) [reply cht-petals/build.sh:7]: just fixed the CI config in #105 so you can [see the error in the logs now too](https://github.com/premAI-io/prem-services/actions/runs/6240447134/job/16940633337?pr=104)

**Comment by casperdcl** [comment]
Sorry about the force-pushes - was debugging [this questionable GHA CI behaviour](https://github.com/premAI-io/prem-services/pull/106#issuecomment-1726446570) 🤯

File: cht-petals/build.sh (hunk 1 above)
Comment (filopedraz) [reply cht-petals/build.sh:7]: Yep, I provided the wrong model_id

Comment (filopedraz) [reply cht-petals/build.sh:7]: No advantages to build_gpu as of now. It can be an improvement that we do. I ll note it down.

**Comment by casperdcl** [comment]
hmm it seems the `linux/amd64` images won't build?

They result in:

```python
Dockerfile:16
> RUN python3 download.py --model $MODEL_ID
Floating point exception (core dumped)
```

**Comment by filopedraz** [comment]
It's a Petals bug. I will investigate it further. Wanna have the arm images to be done so that I can create a PR to the registry and test the entire flow on Mac :)

**Review by casperdcl — APPROVED** [review]
lgtm - did you wanna add falcon-180b first?

**Comment by filopedraz** [comment]
Falcon requires an HF token in order to fetch the weights due to the terms of services. I am investigating [here](https://github.com/filopedraz/from-hf-to-torrent) how to upload the weights to Torrent and then download them with a CLI that we can use inside a dockerfile. :)

**Comment by casperdcl** [comment]
... or just add the HF token to GH secrets :)

**Pull request merged**